torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -282,6 +282,62 @@ void setstateTuple(
     }
   }
 }
+
+// Check if the global static map of backend debug info
+// contains debug info for this module and any of its children.
+// If so combine all the maps together and return one.
+DelegateDebugInfoMapType getBackendDebugInfoMap(const Module& m) {
+  DelegateDebugInfoMapType debug_map;
+  const auto& map =
+      getStaticBackendModuleDebugInfoMapPtr()->getDebugInfoMap(m._ivalue());
+  if (map) {
+    debug_map.insert(map.value().begin(), map.value().end());
+  }
+  for (const auto& c : m.children()) {
+    const auto& child_debug_map = getBackendDebugInfoMap(c);
+    debug_map.insert(child_debug_map.begin(), child_debug_map.end());
```

Comment:
Would it be faster if you had one top-level map that you passed by reference into recursive calls and added all items to instead of aggregating a bunch of maps?

Reply:
Yes, let me do that.